### ISSUE TITLE
Invalid JSON in state.json

### ISSUE THREAD

**Issue by sopoforic** [burtonator/polar-bookshelf#854]
One of my documents has invalid JSON in its `state.json`. It appears that the last fourteen lines have been duplicated:

```
    "152": {
      "pagemarks": {},
      "notes": {},
      "comments": {},
      "questions": {},
      "flashcards": {},
      "textHighlights": {},
      "areaHighlights": {},
      "screenshots": {},
      "thumbnails": {},
      "readingProgress": {},
      "pageInfo": {
        "num": 152
      }
    }
  }
}    "comments": {},
      "questions": {},
      "flashcards": {},
      "textHighlights": {},
      "areaHighlights": {},
      "screenshots": {},
      "thumbnails": {},
      "readingProgress": {},
      "pageInfo": {
        "num": 152
      }
    }
  }
}
```

The [rest of the file](https://github.com/burtonator/polar-bookshelf/files/3100929/state.json.zip) appears to be valid.

I presume the corruption occurred several days ago:

```
    "readingPerDay": {
      "2019-04-03": 1,
      "2019-04-04": 0.17,
      "2019-04-12": 0.65,
      "2019-04-05": 1.46,
      "2019-04-18": 1.12
    }
```

Manually deleting the extra data from the end of `state.json` appears to have corrected the problem.

**Desktop (please complete the following information):**
 - OS: Windows 10 x64
 - Version: 1.17.4

**Comment by sopoforic** [comment]
This could be related to bug #811.

**Comment by burtonator** [comment]
ok.. thanks!.. I should have a fix for this out soon.  I implemented better atomic writes but they broke on windows so I need to figure out why they're broken there.

**Comment by burtonator** [comment]
My windows branch gives me: 

```
[error] [1.18.0] [MainAppExceptionHandlers.js] Unhandled rejection:  { EPERM: operation not permitted, rename 'C:\Users\admin\.polar\6ea16525b2e4eab7b946f68419a345a6\.state.json-12125' -> 'C:\Users\admin\.polar\6ea16525b2e4eab7b946f68419a345a6\state.json'
    at Function.<anonymous> (C:\Users\admin\polar-bookshelf\web\js\util\Files.ts:563:24)
    at Generator.next (<anonymous>)
    at C:\Users\admin\polar-bookshelf\web\js\util\Files.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\admin\polar-bookshelf\web\js\util\Files.js:3:12)
    at Function.withProperException (C:\Users\admin\polar-bookshelf\web\js\util\Files.js:381:16)
    at Function.<anonymous> (C:\Users\admin\polar-bookshelf\web\js\util\Files.ts:483:21)
    at Generator.next (<anonymous>)
    at C:\Users\admin\polar-bookshelf\web\js\util\Files.js:7:71
    at new Promise (<anonymous>)
  errno: -4048,
  code: 'EPERM',
  syscall: 'rename',
  path:
   'C:\\Users\\admin\\.polar\\6ea16525b2e4eab7b946f68419a345a6\\.state.json-12125',
  dest:
   'C:\\Users\\admin\\.polar\\6ea16525b2e4eab7b946f68419a345a6\\state.json' }
```

... this might have just been a permissions error on MY machine.. try to scrap and do this from default and see what happens.

one major problem though is that the exception is NOT shown to the user.

**Comment by burtonator** [comment]
I should try to duplicate this and figure out if I can get the exception passed to the user.

**Comment by burtonator** [comment]
seems to have only been one file on my machine and also after a new install it works...

**Comment by burtonator** [comment]
closign this as it will be resolved in 1.18...
